### PR TITLE
Added extraction of messages from EvolutionBatch when decoding RawMessages

### PR DESCRIPTION
**Pull request by Max-81**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -194,6 +194,9 @@ The filtering can also be applied for pins with  `subscribe` attribute.
 
 ## Release notes
 
++ 3.7.2
+    + Added extraction of messages from EvolutionBatch when decoding RawMessages
```
Comment (Nikita-Smirnov-Exactpro) [inline README.md:198]: Please information about cases where this change can help

**Review by Nikita-Smirnov-Exactpro — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/exactpro/th2/codec/DecodeProcessor.kt [hunk 2]
```diff
@@ -40,7 +41,17 @@ class DecodeProcessor(
...
+            val decodedMsg = getCodec().decode(data, source.toCodecContext())
+
+            val decodedMessages = arrayListOf<IMessage>()
+            decodedMsg.forEach { msg ->
+                if (msg.name == EvolutionBatch.MESSAGE_NAME) {
+                    decodedMessages += EvolutionBatch(msg).batch
+                } else {
+                    decodedMessages += msg
+                }
+            }
```
Comment (OptimumOpium) [inline src/main/kotlin/com/exactpro/th2/codec/DecodeProcessor.kt:53]: Can be replaced with the following code
```suggestion
            val decodedMessages = getCodec().decode(data, source.toCodecContext())
                            .flatMap {
                              if (msg.name == EvolutionBatch.MESSAGE_NAME) {
                                  EvolutionBatch(msg).batch
                              } else {
                                  listOf(msg)
                              }
                            }
```

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -194,6 +194,9 @@ The filtering can also be applied for pins with  `subscribe` attribute.
 
 ## Release notes
 
++ 3.7.2
+    + Added extraction of messages from EvolutionBatch when decoding RawMessages. This is necessary if evolutionSupportEnabled mode is true - Sailfish codecs package the decoding results in EvolutionBatch.
```
Comment (jean-michel-perli-sonsin) [inline README.md:198]:
```suggestion
    + Added extraction of messages from EvolutionBatch when decoding RawMessages. This is necessary if evolutionSupportEnabled mode is set to true - Sailfish codecs package the decoding results in EvolutionBatch.
```

**Review by jean-michel-perli-sonsin — APPROVED** [review]
Hi, please see if you agree with the suggestion made.
The rest of the changes are according to the previous comments/suggestions.
Thanks!

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**